drivers/net/slip.c

```diff
@@ -30,9 +30,16 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <net/net_pkt.h>
 #include <net/net_if.h>
 #include <net/net_core.h>
+#include <net/dummy.h>
```

Comment:
Should this be included conditionally or unconditionally? We could also move it to the else clause of `#if defined(CONFIG_NET_L2_ETHERNET)`. For example:

```C
#if defined(CONFIG_NET_L2_ETHERNET)
#include <net/ethernet.h>
#else
#include <net/dummy.h>
#endif
```

Any thoughts on this?

Reply:
Ethernet L2 and ethernet.h is needed if TAP is enabled (the default). The dummy.h is used used for TUN support but it has limited usability.

Reply:
I am aware of that you didn't answer my question though whether or not `net/dummy.h` should be included conditionally or not.

Reply:
We can include it unconditionally as it just provides struct definition.